src/python/foglamp/device/coap_device.py

```diff
@@ -46,7 +46,7 @@ def plugin_info():
             'interface': '1.0', 'config': _DEFAULT_CONFIG}
 
 
-def plugin_init(config):
+async def plugin_init(config):
```

Comment:
Do all of these need to be async? This is effectively changing the API.

Reply:
@MarkRiddoch I have removed async from defs and have tried to implement the feature _plugin_data in device/server.py/_start(). Please let me know if this is ok.